test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,202 @@
+    client_will_close =
+        absl::EqualsIgnoreCase(response_headers->ProxyConnection()->value().getStringView(),
+                               Http::Headers::get().ConnectionValues.Close);
+  }
+
+  ENVOY_LOG_MISC(trace, "Responded headers {}", response_headers);
+  test_sessions_[0]->stream_response_callbacks_->decodeHeaders(std::move(response_headers), true);
+
+  if (!reuse_connection_ || client_will_close) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream because shouldClose() is true");
+    expectClientCreate(0);
+    expectStreamCreate(0);
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+}
+
+void HealthCheckFuzz::triggerIntervalTimer() {
+  // Interval timer needs to be explicitly enabled, usually by decodeHeaders.
+  if (!test_sessions_[0]->interval_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Interval timer is disabled. Skipping trigger interval timer.");
+    return;
+  }
+  expectStreamCreate(0);
+  ENVOY_LOG_MISC(trace, "Triggered interval timer");
+  test_sessions_[0]->interval_timer_->invokeCallback();
+}
+
+void HealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+  // Timeout timer needs to be explicitly enabled, usually by a call to onIntervalBase().
+  if (!test_sessions_[0]->timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping trigger timeout timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered timeout timer");
+  test_sessions_[0]->timeout_timer_->invokeCallback(); // This closes the client, turns off timeout
+                                                       // and enables interval
+  if (!last_action) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from network timeout");
+    expectClientCreate(0);
+    expectStreamCreate(0);
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+}
+
+void HealthCheckFuzz::raiseEvent(const test::common::upstream::RaiseEvent& event,
+                                 bool last_action) {
+  Network::ConnectionEvent eventType;
+  switch (event) {
+  case test::common::upstream::RaiseEvent::CONNECTED: {
+    eventType = Network::ConnectionEvent::Connected;
+    break;
+  }
+  case test::common::upstream::RaiseEvent::REMOTE_CLOSE: {
+    eventType = Network::ConnectionEvent::RemoteClose;
+    break;
+  }
+  case test::common::upstream::RaiseEvent::LOCAL_CLOSE: {
+    eventType = Network::ConnectionEvent::LocalClose;
+    break;
+  }
+  default: // shouldn't hit
+    eventType = Network::ConnectionEvent::Connected;
+    break;
+  }
+
+  switch (type_) {
+  case HealthCheckFuzz::Type::HTTP: {
+    test_sessions_[0]->client_connection_->raiseEvent(eventType);
+    if (!last_action && eventType != Network::ConnectionEvent::Connected) {
+      ENVOY_LOG_MISC(trace, "Creating client and stream from close event");
+      expectClientCreate(0);
+      expectStreamCreate(0);
+      test_sessions_[0]->interval_timer_->invokeCallback();
```

Comment:
another interval timer -- I'm wondering, maybe you want `triggerIntervalTimer` to take in a boolean for expectClientCreate and just call it in all three of these places?